feature/policy_forwarding/encapsulation/otg_tests/decap_gre_ipv4/decap_gre_ipv4_test.go

```diff
@@ -0,0 +1,836 @@
+				mpls ip
+				mpls static top-label %v %s %s pop payload-type ipv4
+				`, LBL1, dut.Port(t, "port2").Name(), otgPort2.IPv4,
+			)
+
+			gpbSetRequest := buildCliConfigRequest(jsonConfig)
+
+			if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+				t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+			}
+		default:
+			t.Errorf("Deviation StaticMplsLspUnsupported is not handled for the dut: %v", dut.Vendor())
+		}
+	} else {
+		d := &oc.Root{}
+		fptest.ConfigureDefaultNetworkInstance(t, dut)
+		mplsCfg := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateMpls()
+		staticMplsCfg := mplsCfg.GetOrCreateLsps().GetOrCreateStaticLsp("lsp1")
+		staticMplsCfg.GetOrCreateEgress().SetIncomingLabel(oc.UnionUint32(LBL1))
+		staticMplsCfg.GetOrCreateEgress().SetNextHop(otgPort2.IPv4)
+		staticMplsCfg.GetOrCreateEgress().SetPushLabel(oc.Egress_PushLabel_IMPLICIT_NULL)
+
+		gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Mpls().Config(), mplsCfg)
+	}
+}
+
+// GRE traffic with destination DECAP-DST using destination-address-prefix-set and decapsulate.
+// Adding deviation, currently gre decapsulation is not supported, using CLI.
+func configureGRETunnel(t *testing.T, dut *ondatra.DUTDevice) {
+	if deviations.GreDecapsulationUnsupported(dut) {
+		switch dut.Vendor() {
+		case ondatra.ARISTA:
+			gnmiClient := dut.RawAPIs().GNMI(t)
+			jsonConfig := fmt.Sprintf(`
+			ip decap-group %s
+			 tunnel type gre
+			 tunnel decap-ip %s
+			`, decapGrpName, strings.Split(decapDesIpv4IP, "/")[0])
+			gpbSetRequest := buildCliConfigRequest(jsonConfig)
+			if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+				t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+			}
+		default:
+			t.Errorf("Deviation GreDecapsulationUnsupported is not handled for the dut: %v", dut.Vendor())
+		}
+	} else {
+		d := &oc.Root{}
+		ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+		ni1.SetType(oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+		npf := ni1.GetOrCreatePolicyForwarding()
+		np := npf.GetOrCreatePolicy("PBR-MAP")
+		np.PolicyId = ygot.String("PBR-MAP")
+		np.Type = oc.Policy_Type_PBR_POLICY
+
+		npRule := np.GetOrCreateRule(10)
+		ip := npRule.GetOrCreateIpv4()
+		ip.DestinationAddressPrefixSet = ygot.String(decapDesIpv4IP)
+		npAction := npRule.GetOrCreateAction()
+		npAction.DecapsulateGre = ygot.Bool(true)
+
+		port := dut.Port(t, "port1")
+		ingressPort := port.Name()
+		t.Logf("Applying forwarding policy on interface %v ... ", ingressPort)
+
+		intf := npf.GetOrCreateInterface(ingressPort)
+		intf.ApplyForwardingPolicy = ygot.String("PBR-MAP")
+		intf.GetOrCreateInterfaceRef().Interface = ygot.String(ingressPort)
+
+		gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Config(), ni1)
+	}
+}
+
+func buildCliConfigRequest(config string) *gpb.SetRequest {
```

Comment:
avoid cut and paste of code, instead use the helper:

https://github.com/openconfig/featureprofiles/blob/5bd6746949359f4a3851e735c9773b040909b65a/internal/helpers/helpers.go#L141